test/unittests/test_builtins.cpp

```diff
@@ -65,3 +65,34 @@ TEST(builtins, clz_zero)
     EXPECT_EQ(clz(uint64_t{0}), 64u);
     EXPECT_EQ(clz_generic(uint64_t{0}), 64u);
 }
+
+namespace
+{
+constexpr int func() noexcept
+{
+    return is_constant_evaluated() ? 1 : 2;
+}
+}  // namespace
+
+TEST(builtins, is_constant_evaluated)
+{
+    constexpr auto a = is_constant_evaluated();
+    EXPECT_TRUE(a);
+
+    auto b = is_constant_evaluated();
+#if __has_builtin(__builtin_is_constant_evaluated)
```

Comment:
Instead of repeating `__has_builtin()` check here, I'd check compiler version and add expectations for some where it should work and some where it should not work.